### PR TITLE
License expression changes in registration

### PR DESCRIPTION
**Pull request by agr**
Should be OK to merge now.

https://github.com/NuGet/Engineering/issues/1787

### CODE REVIEW

**Comment by karann-msft** [comment]
OPS Build status updates of commit _[9952ee7](https://github.com/NuGet/docs.microsoft.com-nuget/pull/1220/commits/9952ee730273b8e02b13f163dfbbe845fed38b0d)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/api/registration-base-url-resource.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/agr-licensing/docs/api/registration-base-url-resource.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/registration-base-url-resource?branch=pr-en-us-1220) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C12%5C12%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201812120105363959-1220%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=H8AZMq4T%2Fb5chTWNw6S9jSPa0%2Bd0sFyhwqga4qF4WBo%3D&st=2018-12-12T01%3A02%3A47Z&se=2019-01-12T01%3A07%3A47Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/api/registration-base-url-resource.md [hunk 1]
```diff
@@ -181,6 +182,13 @@ The `dependencyGroups` property is an array of objects representing the dependen
 framework. If the package has no dependencies, the `dependencyGroups` property is missing, an empty array, or the
 `dependencies` property of all groups is empty or missing.
 
+The `licenseExpression` property may only have value specified if the package manifest
```
Comment (joelverhagen) [inline docs/api/registration-base-url-resource.md]: have **a** value specified

File: docs/api/registration-base-url-resource.md [hunk 2]
```diff
@@ -181,6 +182,13 @@ The `dependencyGroups` property is an array of objects representing the dependen
...
+contains `&lt;license type='expression'&gt;` element. The value of the `licenseExpression` property complies with
```
Comment (joelverhagen) [inline docs/api/registration-base-url-resource.md]: `<` and `>`. See the preview. This is verbatim.

File: docs/api/registration-base-url-resource.md [hunk 3]
```diff
@@ -181,6 +182,13 @@ The `dependencyGroups` property is an array of objects representing the dependen
...
+[SPDX license expression syntax](https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60).
+
+The `licenseExpression` property may be an empty string or property may not exist under `catalogEntry`. In both
```
Comment (joelverhagen) [inline docs/api/registration-base-url-resource.md]: Really, it can be an empty string?

File: docs/api/registration-base-url-resource.md [hunk 4]
```diff
@@ -166,7 +166,8 @@ dependencyGroups         | array of objects           | no       | The dependenc
 description              | string                     | no       | 
 iconUrl                  | string                     | no       | 
 id                       | string                     | yes      | The ID of the package
-licenseUrl               | string                     | no       | 
+licenseUrl               | string                     | no       |
+licenseExpression        | string                     | no       | The license expression specified in the package manifest.
```
Comment (joelverhagen) [inline docs/api/registration-base-url-resource.md]: `manifest.` -> `manifest`

Comment (joelverhagen) [reply docs/api/registration-base-url-resource.md]: I don't think it's necessary to make this note. None of the other fields have it and it's kind of an implementation detail about where it came from. The point is, the server considers this package to be licensed using that SPDX expression.

File: docs/api/registration-base-url-resource.md [hunk 5]
```diff
@@ -181,6 +182,13 @@ The `dependencyGroups` property is an array of objects representing the dependen
 framework. If the package has no dependencies, the `dependencyGroups` property is missing, an empty array, or the
 `dependencies` property of all groups is empty or missing.
 
+The `licenseExpression` property may only have value specified if the package manifest
+contains `&lt;license type='expression'&gt;` element. The value of the `licenseExpression` property complies with
+[SPDX license expression syntax](https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60).
+
+The `licenseExpression` property may be an empty string or property may not exist under `catalogEntry`. In both
+cases it is an indication that the original package did not have a license expression specified.
```
Comment (joelverhagen) [inline docs/api/registration-base-url-resource.md]: Can we get a link to the client-side/pack docs regarding license expression?

**Comment by karann-msft** [comment]
OPS Build status updates of commit _[cb1dbcb](https://github.com/NuGet/docs.microsoft.com-nuget/pull/1220/commits/cb1dbcbd1dc401d7990aec85765a2fc4fd8adadc)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/api/registration-base-url-resource.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/agr-licensing/docs/api/registration-base-url-resource.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/api/registration-base-url-resource?branch=pr-en-us-1220) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C12%5C15%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201812150035380785-1220%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=peJ20OHM6Xl31L0xopBcw5hMui3uUB7ozO3xeFceCvU%3D&st=2018-12-15T00%3A31%3A38Z&se=2019-01-15T00%3A36%3A38Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**